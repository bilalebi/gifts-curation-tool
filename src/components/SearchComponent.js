import React from 'react';
import PropTypes from 'prop-types';
import ResultsComponent from "./ResultsComponent";

class SearchComponent extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            value: this.props.value,
            showResults: false
        };
        this.handleClick = this.handleClick.bind(this);
    }

    render() {

        return (
            <div>
                Current search value: {this.state.value}
                <div className="input-group">
                    <input
                        value={this.state.value}
                        type="text"
                        placeholder="ENST00000620613, A7E2Y1, ..."
                        className="input-group-field"
                        onChange={ this.handleChange }/>
                    <div className="input-group-button">
                        <button type="button" className="button" onClick={this.handleClick}>Submit</button>
                    </div>
                </div>
            </div>
        );
    }

    handleChange = (e) => {
        this.setState({value :e.target.value});
    }

    handleClick = () => {
        this.props.action(this.state.value);
    }

}

SearchComponent.propTypes = {
    value: PropTypes.string,
};

export default SearchComponent;