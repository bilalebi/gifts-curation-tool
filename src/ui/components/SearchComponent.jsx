import React from 'react';
import { withRouter } from 'react-router-dom';

class SearchComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      searchTerm: props.searchTerm ? props.searchTerm : '',
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleChange = (e) => {
    this.setState({ searchTerm: e.target.value });
  };

  handleClick = () => {
    this.props.updateSearchTerm(this.state.searchTerm);
    // this.props.history.push({
    //   pathname: '/mappings',
    //   search: `?searchTerm=${this.state.searchTerm}`,
    // });
  };

  render() {
    return (
      <div>
        <div className="input-group">
          <input
            value={this.state.searchTerm}
            type="text"
            placeholder="ENST00000620613, A7E2Y1, ..."
            className="input-group-field"
            onChange={this.handleChange}
          />
          <div className="input-group-button">
            <button type="button" className="button" onClick={this.handleClick}>
              Submit
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default withRouter(SearchComponent);
