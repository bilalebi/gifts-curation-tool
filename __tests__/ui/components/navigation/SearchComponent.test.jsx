import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import SearchComponent from '../../../../src/ui/components/SearchComponent';

describe('SearchComponent component', () => {
  test('should render', () => {
    const component = renderer
      .create(<MemoryRouter>
        <SearchComponent />
              </MemoryRouter>)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
