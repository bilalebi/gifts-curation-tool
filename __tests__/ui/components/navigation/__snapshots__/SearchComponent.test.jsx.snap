// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent component should render 1`] = `
<div>
  <div
    className="input-group"
  >
    <input
      className="input-group-field"
      onChange={[Function]}
      placeholder="ENST00000620613, A7E2Y1, ..."
      type="text"
      value=""
    />
    <div
      className="input-group-button"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;
